### PR TITLE
set all elements to aria-hidden when date modal is active

### PR DESCRIPTION
**Pull request by BobanL**
# FRONTEND PULL REQUEST

## Related Issue

- #4075 
- When a modal is active, everything else should have the aria-hidden attribute set to true.

## Changes Proposed

- Add a mutation observer to the sign up page. That observer will look for the date picker to become "unhidden" and then update the modal active state which gets propagated to all elements.

## Additional Information

- For the sake of this PR's size. Only the identity verification date modal has been complete. Other PRs will be open after completed.
- The only element that has not been hidden is the calendar button used to activate the modal itself since there are no attributes that can be passed to it based on the library we use. 

## Testing

- Go to signup on [dev4](https://dev4.simplereport.gov/app/sign-up), fill out that information to get to the identity verification page. Check to make sure all elements are hidden when the date of birth modal is active.
 
## Screenshots / Demos

- image
-  image

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [ ] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [ ] Any content changes have been approved by content team

### Support
- [ ] Any changes that might generate new support requests have been flagged to the support team
- [ ] Any changes to support infrastructure have been demo'd to support team

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### Documentation
- [ ] Any changes to the startup configuration have been documented in the README

### CODE REVIEW

**Review by fzhao99 — DISMISSED** [review]
TIL about mutation observers! Very cool pattern and very useful for this use case.

Great work on this Boban! Happy to help propagating this to other modals if you want some help splitting up the work.

**Review by emyl3 — APPROVED** [review]
This looks great! Thank you for adding all the tests as well! 🤩

File: frontend/src/app/commonComponents/Button/Button.test.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  });
+  describe("ariaDescribedBy", () => {
+    it("should set aria-describedby", () => {
+      render(<Button ariaDescribedBy={"someLabel"} />);
+      expect(screen.getByRole("button")).toHaveAttribute(
+        "aria-describedby",
+        "someLabel"
+      );
+    });
+    // can we get rid of the or o
```
Comment (BobanL) [inline frontend/src/app/commonComponents/Button/Button.test.tsx:95]:
```suggestion
```

File: frontend/src/app/commonComponents/Button/Button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      render(<Button />);
+      expect(screen.getByRole("button")).toHaveClass("usa-button", {
+        exact: true,
+      });
+    });
+    it("should include usa-button--variant", () => {
+      render(<Button variant={"big"} />);
+      expect(screen.getByRole("button")).toHaveClass("usa-button--big");
+    });
+    it("should include primary", () => {
```
Comment (BobanL) [inline frontend/src/app/commonComponents/Button/Button.test.tsx:17]:
```suggestion
    it("should include class given", () => {
```

**Review by fzhao99 — APPROVED** [review]

**Review by emyl3 — APPROVED** [review]

**Review by emmastephenson — APPROVED** [review]
This looks great, thank you for the through test coverage!

**Review by georgehager — APPROVED** [review]
Nice! Love the additional tests. 🙌

**Pull request merged**